### PR TITLE
Add GQL sync device functionality to sdk

### PR DESCRIPTION
**Pull request by adrianaudienceview**

### CODE REVIEW

File: sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt [hunk 1]
```diff
@@ -1,48 +1,79 @@
...
+import com.apollographql.apollo.ApolloCall
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.api.toInput
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.exception.ApolloHttpException
+import com.realifetech.SyncDeviceMutation
+import com.realifetech.sdk.RealifeTech
 import com.realifetech.sdk.core.data.database.preferences.configuration.ConfigurationStorage
+import com.realifetech.sdk.core.data.model.basket.Basket
```
Comment (rossp-av) [inline sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt:12]: Why is this needed?

File: sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt [hunk 2]
```diff
@@ -1,48 +1,79 @@
...
         }
     }
 
     private companion object {
         private const val ANDROID = "ANDROID"
     }
-}
\ No newline at end of file
+}
+
```
Comment (rossp-av) [inline sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt:66]: Can you remove the extra lines?

File: sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt (hunk 1 above)
Comment (adrianaudienceview) [reply sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt:12]: I´ll remove it

File: sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt (hunk 2 above)
Comment (adrianaudienceview) [reply sdk/src/main/java/com/realifetech/sdk/general/data/DeviceNetworkDataSourceImpl.kt:66]: I'll remove it

**Review by rossp-av — APPROVED** [review]
